teuthology/orchestra/daemon/state.py

```diff
@@ -103,11 +103,14 @@ def signal(self, sig, silent=False):
         :param sig: signal to send
         """
         if self.running():
-            self.proc.stdin.write(struct.pack('!b', sig))
+            try:
+                self.proc.stdin.write(struct.pack('!b', sig))
+            except:
```

Comment:
`except IOError as e:`

(blanket excepts are discouraged!)

Reply:
Fixed the except. 

But I was wondering if there is some way to check stdin is still open - like `if (self.proc.stdin)`, so that I could avoid trying to write the SIGTERM.

Reply:
try/except would be more idiomatic I think.